sklearn/ensemble/_weight_boosting.py

```diff
@@ -111,7 +111,7 @@ def fit(self, X, y, sample_weight=None):
         sample_weight = _check_sample_weight(sample_weight, X, np.float64)
         sample_weight /= sample_weight.sum()
         if np.any(sample_weight < 0):
-            raise ValueError("sample_weight cannot contain negative weights")
+            raise ValueError("There are negative values in sample_weight")
```

Comment:
seems like this line is never run in the tests, which probably means you need to have a separate test for it. I think even if the user sets the flag, this model still requires positive weights, and that's what needs to be tested.